rotkehlchen/chain/ethereum/modules/convex/decoder.py

```diff
@@ -0,0 +1,131 @@
+                    else:
+                        event.notes = f'Spent {event.balance.amount} {event.asset.symbol} as part of withdrawal from convex'  # noqa: E501
+                else:
+                    event.event_type = HistoryEventType.DEPOSIT
+                    event.counterparty = CPT_CONVEX
+                    if tx_log.address in CONVEX_POOLS:
+                        event.notes = f'Deposit {event.balance.amount} {event.asset.symbol} into convex {CONVEX_POOLS[tx_log.address]} pool'  # noqa: E501
+                    else:
+                        event.notes = f'Deposit {event.balance.amount} {event.asset.symbol} into convex'  # noqa: E501
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE
+            ):
+                if tx_log.topics[0] in WITHDRAWAL_TOPICS:
+                    event.event_type = HistoryEventType.WITHDRAWAL
+                    if tx_log.address in CONVEX_POOLS:
+                        event.notes = f'Withdraw {event.balance.amount} {event.asset.symbol} from convex {CONVEX_POOLS[tx_log.address]} pool'  # noqa: E501
+                    else:
+                        event.notes = f'Withdraw {event.balance.amount} {event.asset.symbol} from convex'  # noqa: E501
+                    event.counterparty = CPT_CONVEX
+                elif tx_log.topics[0] in REWARD_TOPICS:
+                    event.event_subtype = HistoryEventSubType.REWARD
+                    event.counterparty = CPT_CONVEX
+                    if tx_log.address in CONVEX_POOLS:
+                        event.notes = f'Receive reward {event.balance.amount} {event.asset.symbol} from convex {CONVEX_POOLS[tx_log.address]} pool'  # noqa: E501
+                    else:
+                        event.notes = f'Receive reward {event.balance.amount} {event.asset.symbol} from convex'  # noqa: E501
+        return None, None
+
+    @staticmethod
+    def _maybe_enrich_convex_transfers(
+            token: EthereumToken,  # pylint: disable=unused-argument
+            tx_log: EthereumTxReceiptLog,
+            transaction: EthereumTransaction,
+            event: HistoryBaseEntry,
+            action_items: List[ActionItem],  # pylint: disable=unused-argument
+    ) -> bool:
+        if (
+            tx_log.topics[0] == TRANSFER_TOPIC and
+            tx_log.topics[1] in CONVEX_ABRAS_HEX and
+            event.location_label == transaction.from_address and
+            event.event_type == HistoryEventType.RECEIVE and
+            event.event_subtype == HistoryEventSubType.NONE
+        ):
+            event.event_subtype = HistoryEventSubType.REWARD
+            if tx_log.address in CONVEX_POOLS:
+                event.notes = f'Receive reward {event.balance.amount} {event.asset.symbol} from convex {CONVEX_POOLS[tx_log.address]} pool'  # noqa: E501
```

Comment:
same here for rewording. And a question. Why do you do the rewards detection both here and above? What is the differene?

Reply:
Used for rewards paid with abracadabras. Problem is that the transfer event in this case happens at the end of the transaction and there is no reward event after it to emit event processing.

Put same text to the docstring